chips/stm32f429zi/src/ethernet/ethernet.rs

```diff
@@ -0,0 +1,2288 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright 2023 OxidOS Automotive SRL
+//
+// Author: Ioan-Cristian CÎRSTEA <ioan.cirstea@oxidos.io>
+#![deny(missing_docs)]
+#![deny(dead_code)]
+#![deny(unused_imports)]
+//! Ethernet firmware for STM32F429ZI
+//!
+//! # Usage
+//!
+//! The Ethernet requires that AHB frequency must be at least 25MHz. Default boot configuration
+//! sets up a 16MHz clock frequency. Nucleo-F429ZI already contains code to setup the Ethernet.
+//! Uncomment code sections marked with ETHERNET to use this peripheral.
+
+use core::cell::Cell;
+
+use cortexm4::support::nop;
+use kernel::hil::ethernet::Configure;
+use kernel::hil::ethernet::EthernetAdapter;
+use kernel::hil::ethernet::EthernetAdapterClient;
+use kernel::hil::ethernet::EthernetSpeed;
+use kernel::hil::ethernet::MacAddress;
+use kernel::hil::ethernet::OperationMode;
+use kernel::platform::chip::ClockInterface;
+use kernel::utilities::cells::{NumericCellExt, OptionalCell, TakeCell};
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::utilities::StaticRef;
+use kernel::ErrorCode;
+
+use crate::rcc;
+use crate::rcc::PeripheralClock;
+use crate::rcc::PeripheralClockType;
+
+use crate::ethernet::receive_descriptor::ReceiveDescriptor;
+use crate::ethernet::transmit_descriptor::TransmitDescriptor;
+
+register_structs! {
+    /// Ethernet: media access control (MAC)
+    Ethernet_MacRegisters {
```

Comment:
Nit: usually, names in Rust follow either "snake_case", "UPPER_CASE_SNAKE_CASE" or "PascalCase", but not a mixture of those. Typically we try to stick to those, but here it might not matter: the tock-registers macros generate the appropriate `#[allow(_)]` markers for names to violate these rules (as we use the names for both modules and structs).

Reply:
Fixed names.